packages/block-library/src/search/edit.native.js

```diff
@@ -154,112 +155,165 @@ export default function SearchEdit( {
 		</InspectorControls>
 	);
 
-	const mergeWithBorderStyle = ( style ) => {
-		return { ...style, ...styles.border };
-	};
+	const isButtonInside = buttonPosition === 'button-inside';
 
-	const renderTextField = () => {
-		const inputStyle =
-			buttonPosition === 'button-inside'
-				? styles.searchTextInput
-				: mergeWithBorderStyle( styles.searchTextInput );
+	const borderStyle = usePreferredColorSchemeStyle(
+		styles.border,
+		styles.borderDark
+	);
+
+	const inputStyle = [
+		usePreferredColorSchemeStyle(
+			styles.plainTextInput,
+			styles.plainTextInputDark
+		),
+		! isButtonInside && borderStyle,
+	];
 
+	const placeholderStyle = usePreferredColorSchemeStyle(
+		styles.plainTextPlaceholder,
+		styles.plainTextPlaceholderDark
+	);
+
+	const searchBarStyle = [
+		styles.searchBarContainer,
+		isButtonInside && borderStyle,
+	];
+
+	const renderTextField = () => {
 		return (
-			<PlainText
-				ref={ textInputRef }
-				isSelected={ isPlaceholderSelected }
-				className="wp-block-search__input"
-				style={ inputStyle }
-				numberOfLines={ 1 }
-				ellipsizeMode="tail" // currently only works on ios
-				label={ null }
-				value={ placeholder }
-				placeholder={
-					placeholder ? undefined : __( 'Optional placeholder…' )
-				}
-				onChange={ ( newVal ) =>
-					setAttributes( { placeholder: newVal } )
-				}
-				onFocus={ () => {
-					setIsPlaceholderSelected( true );
-					onFocus();
-				} }
-				onBlur={ () => setIsPlaceholderSelected( false ) }
-			/>
+			<View
+				style={ styles.searchInputContainer }
+				accessible={ true }
+				accessibilityRole="none"
+				accessibilityHint={ __(
+					'Double tap to edit placeholder text'
+				) }
+				accessibilityLabel={ `${ __(
+					'Search input field. Current placeholder value is'
+				) } ${ placeholder }` }
+			>
+				<PlainText
```

Comment:
I thought the same thing but if I add that same code it makes no difference. I think just like the other comment below, that iOS recognizes this component as an input field and forces focus on it's own. I even tried deleting the `componentDidMount` code in the `PlainText` block and tested again and it will still steal the focus the moment the block is activated.